contracts/contracts/vault/VaultAdmin.sol

```diff
@@ -0,0 +1,270 @@
+    }
+
+    /**
+     * @notice Set the weights for multiple strategies.
+     * @param _strategyAddresses Array of strategy addresses
+     * @param _weights Array of corresponding weights, with 18 decimals.
+     *                 For ex. 100%=1e18, 30%=3e17.
+     */
+    function setStrategyWeights(
+        address[] calldata _strategyAddresses,
+        uint256[] calldata _weights
+    ) external onlyGovernor {
+        require(
+            _strategyAddresses.length == _weights.length,
+            "Parameter length mismatch"
+        );
+
+        for (uint256 i = 0; i < _strategyAddresses.length; i++) {
+            strategies[_strategyAddresses[i]].targetWeight = _weights[i];
+        }
+
+        emit StrategyWeightsUpdated(_strategyAddresses, _weights);
+    }
+
+    /***************************************
+                    Pause
+    ****************************************/
+
+    /**
+     * @dev Set the deposit paused flag to true to prevent rebasing.
+     */
+    function pauseRebase() external onlyGovernor {
+        rebasePaused = true;
+    }
+
+    /**
+     * @dev Set the deposit paused flag to true to allow rebasing.
+     */
+    function unpauseRebase() external onlyGovernor {
+        rebasePaused = false;
+    }
+
+    /**
+     * @dev Set the deposit paused flag to true to prevent deposits.
+     */
+    function pauseDeposits() external onlyGovernor {
+        depositPaused = true;
+
+        emit DepositsPaused();
+    }
+
+    /**
+     * @dev Set the deposit paused flag to false to enable deposits.
+     */
+    function unpauseDeposits() external onlyGovernor {
+        depositPaused = false;
+
+        emit DepositsUnpaused();
+    }
+
+    /***************************************
+                    Rewards
+    ****************************************/
+
+    /**
+     * @dev Transfer token to governor. Intended for recovering tokens stuck in
+     *      contract, i.e. mistaken sends.
+     * @param _asset Address for the asset
+     * @param _amount Amount of the asset to transfer
+     */
+    function transferToken(address _asset, uint256 _amount)
+        external
+        onlyGovernor
+    {
+        IERC20(_asset).transfer(governor(), _amount);
+    }
+
+    /**
+     * @dev Collect reward tokens from all strategies.
+     */
+    function collectRewardTokens() external onlyGovernor {
+        for (uint256 i = 0; i < allStrategies.length; i++) {
+            collectRewardTokens(allStrategies[i]);
+        }
+    }
+
+    /**
+     * @dev Collect reward tokens from a single strategy and transfer them to
+            Vault.
+     * @param _strategyAddr Address of the strategy to collect rewards from
+     */
+    function collectRewardTokens(address _strategyAddr) public onlyGovernor {
+        IStrategy strategy = IStrategy(_strategyAddr);
+        strategy.collectRewardToken();
+    }
+
+    /***************************************
+                    Pricing
```

Comment:
Those pricing methods are not for governor only so why putting them here?

Reply:
To save more space on the Core

Reply:
Makes sense.

It's kinda weird to call this contract VaultAdmin then if it includes non admin related methods that are called by the DApp... But that's just a naming thing. Maybe we could call it "VaultUtils"? I'm terrible at namig so I let you guys make that very difficult decision :)